client/state/sites/domains/selectors.js

```diff
@@ -65,12 +65,14 @@ export function getDecoratedSiteDomains( state, siteId ) {
 		return {
 			...domain,
 
-			// Add registration moment from registrationDate
```

Comment:
Isn't this whole chunk here to avoid storing `moment` dates? This can probably just go away, because as far as I can see, you won't have any of the `...Date` variables here?